app/src/main/java/org/mozilla/fenix/settings/account/AccountStore.kt

```diff
@@ -0,0 +1,53 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+   License, v. 2.0. If a copy of the MPL was not distributed with this
+   file, You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+package org.mozilla.fenix.settings.account
+
+import mozilla.components.lib.state.Action
+import mozilla.components.lib.state.State
+import mozilla.components.lib.state.Store
+
+/**
+ * The [Store] for holding the [AccountSettingsState] and applying [AccountAction]s.
+ */
+class AccountSettingsStore(
+    initialState: AccountSettingsState
+) : Store<AccountSettingsState, AccountSettingsAction>(
+    initialState,
+    ::accountStateReducer
+)
+
+sealed class LastSyncTime {
+    object Never : LastSyncTime()
+    data class Failed(val lastSync: Long) : LastSyncTime()
+    data class Success(val lastSync: Long) : LastSyncTime()
+}
+
+/**
+ * The state for the Account Settings Screen
+ */
+data class AccountSettingsState(
+    val lastSyncedDate: LastSyncTime,
```

Comment:
nit: might make sense to give this default values?

Reply:
@csadilek Should all states be given default values? I noticed that classes like SearchState also don't have default values.

Reply:
@jyeontaek it depends ;). Really just a nit, as I thought it makes it easier for you to set an initial state esp. in tests. We do that in A-C: https://github.com/mozilla-mobile/android-components/blob/master/components/browser/state/src/main/java/mozilla/components/browser/state/state/BrowserState.kt#L17

Reply:
Feel free to skip adding defaults. I think we can merge this PR once we have the `AccountSettingsStoreTest` :).